PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:11 + add:0, change matches:
-    _inherit = ['calendar.event']
-    _inherit = ['digest.digest']
-    _inherit = ['hr.department']
-    _inherit = ["hr.employee"]
-    _inherit = ['ir.ui.menu']
-    _inherit = ['mail.activity.plan']
-    _inherit = ["res.company"]
-    _inherit = ['res.config.settings']
-    _inherit = ['res.users']
-    _inherit = ['utm.campaign']
-    _inherit = ['utm.source']

Total Changes: 31

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_recruitment/models/calendar.py
+++ b/addons/hr_recruitment/models/calendar.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class CalendarEvent(models.Model):
-    _inherit = ['calendar.event']
+    _inherit = 'calendar.event'
 
     @api.model
     def default_get(self, fields):

--- a/addons/hr_recruitment/models/digest.py
+++ b/addons/hr_recruitment/models/digest.py
@@ -6,7 +6,7 @@ from odoo.exceptions import AccessError
 
 
 class DigestDigest(models.Model):
-    _inherit = ['digest.digest']
+    _inherit = 'digest.digest'
 
     kpi_hr_recruitment_new_colleagues = fields.Boolean('New Employees')
     kpi_hr_recruitment_new_colleagues_value = fields.Integer(compute='_compute_kpi_hr_recruitment_new_colleagues_value')

--- a/addons/hr_recruitment/models/hr_applicant.py
+++ b/addons/hr_recruitment/models/hr_applicant.py
@@ -20,6 +20,7 @@ AVAILABLE_PRIORITIES = [
 
 
 class HrApplicant(models.Model):
+    _name = 'hr.applicant'
     _description = "Applicant"
     _order = "priority desc, id desc"
     _inherit = ['mail.thread.cc',

--- a/addons/hr_recruitment/models/hr_applicant_category.py
+++ b/addons/hr_recruitment/models/hr_applicant_category.py
@@ -7,6 +7,7 @@ from odoo import fields, models
 
 
 class HrApplicantCategory(models.Model):
+    _name = 'hr.applicant.category'
     _description = "Category of applicant"
 
     def _get_default_color(self):

--- a/addons/hr_recruitment/models/hr_applicant_refuse_reason.py
+++ b/addons/hr_recruitment/models/hr_applicant_refuse_reason.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class HrApplicantRefuseReason(models.Model):
+    _name = 'hr.applicant.refuse.reason'
     _description = 'Refuse Reason of Applicant'
     _order = 'sequence'
 

--- a/addons/hr_recruitment/models/hr_candidate.py
+++ b/addons/hr_recruitment/models/hr_candidate.py
@@ -10,6 +10,7 @@ from odoo.osv import expression
 
 
 class HrCandidate(models.Model):
+    _name = 'hr.candidate'
     _description = "Candidate"
     _inherit = ['mail.thread.cc',
                'mail.thread.main.attachment',

--- a/addons/hr_recruitment/models/hr_department.py
+++ b/addons/hr_recruitment/models/hr_department.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class HrDepartment(models.Model):
-    _inherit = ['hr.department']
+    _inherit = 'hr.department'
 
     new_applicant_count = fields.Integer(
         compute='_compute_new_applicant_count', string='New Applicant', compute_sudo=True)

--- a/addons/hr_recruitment/models/hr_employee.py
+++ b/addons/hr_recruitment/models/hr_employee.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class HrEmployee(models.Model):
-    _inherit = ["hr.employee"]
+    _inherit = "hr.employee"
 
     # YTI Rename into candidate_ids
     candidate_id = fields.One2many('hr.candidate', 'employee_id', 'Candidate', groups="hr.group_hr_user")

--- a/addons/hr_recruitment/models/hr_job.py
+++ b/addons/hr_recruitment/models/hr_job.py
@@ -10,6 +10,7 @@ from odoo.tools.convert import convert_file
 
 
 class HrJob(models.Model):
+    _name = 'hr.job'
     _inherit = ["mail.alias.mixin", "hr.job"]
     _order = "sequence, name asc"
 

--- a/addons/hr_recruitment/models/hr_job_platform.py
+++ b/addons/hr_recruitment/models/hr_job_platform.py
@@ -5,6 +5,7 @@ from odoo.tools import email_normalize
 
 
 class HrJobPlatform(models.Model):
+    _name = 'hr.job.platform'
     _description = 'Job Platforms'
 
     name = fields.Char(required=True)

--- a/addons/hr_recruitment/models/hr_recruitment_degree.py
+++ b/addons/hr_recruitment/models/hr_recruitment_degree.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class HrRecruitmentDegree(models.Model):
+    _name = 'hr.recruitment.degree'
     _description = "Applicant Degree"
 
     name = fields.Char("Degree Name", required=True, translate=True)

--- a/addons/hr_recruitment/models/hr_recruitment_source.py
+++ b/addons/hr_recruitment/models/hr_recruitment_source.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class HrRecruitmentSource(models.Model):
+    _name = 'hr.recruitment.source'
     _description = "Source of Applicants"
     _inherit = ['utm.source.mixin']
 

--- a/addons/hr_recruitment/models/hr_recruitment_stage.py
+++ b/addons/hr_recruitment/models/hr_recruitment_stage.py
@@ -5,6 +5,7 @@ from odoo import api, fields, models, _
 
 
 class HrRecruitmentStage(models.Model):
+    _name = 'hr.recruitment.stage'
     _description = "Recruitment Stages"
     _order = 'sequence'
 

--- a/addons/hr_recruitment/models/ir_ui_menu.py
+++ b/addons/hr_recruitment/models/ir_ui_menu.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class IrUiMenu(models.Model):
-    _inherit = ['ir.ui.menu']
+    _inherit = 'ir.ui.menu'
 
     def _load_menus_blacklist(self):
         res = super()._load_menus_blacklist()

--- a/addons/hr_recruitment/models/mail_activity_plan.py
+++ b/addons/hr_recruitment/models/mail_activity_plan.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class MailActivityPlan(models.Model):
-    _inherit = ['mail.activity.plan']
+    _inherit = 'mail.activity.plan'
 
     def _compute_department_assignable(self):
         super()._compute_department_assignable()

--- a/addons/hr_recruitment/models/res_company.py
+++ b/addons/hr_recruitment/models/res_company.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ["res.company"]
+    _inherit = "res.company"
 
     candidate_properties_definition = fields.PropertiesDefinition('Candidate Properties')
     job_properties_definition = fields.PropertiesDefinition("Job Properties")

--- a/addons/hr_recruitment/models/res_config_settings.py
+++ b/addons/hr_recruitment/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     module_website_hr_recruitment = fields.Boolean(string='Online Posting')
     module_hr_recruitment_survey = fields.Boolean(string='Interview Forms')

--- a/addons/hr_recruitment/models/res_users.py
+++ b/addons/hr_recruitment/models/res_users.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     def _create_recruitment_interviewers(self):
         if not self:

--- a/addons/hr_recruitment/models/utm_campaign.py
+++ b/addons/hr_recruitment/models/utm_campaign.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class UtmCampaign(models.Model):
-    _inherit = ['utm.campaign']
+    _inherit = 'utm.campaign'
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_utm_campaign_job(self):

--- a/addons/hr_recruitment/models/utm_source.py
+++ b/addons/hr_recruitment/models/utm_source.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class UtmSource(models.Model):
-    _inherit = ['utm.source']
+    _inherit = 'utm.source'
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_linked_recruitment_sources(self):
